BB: Allow API Usage in BN 6 without 7 - Grants Augment for BN 7 completion

Sphyxis opened the pull request:
This allows the user to use the Netscript API in BN 6 without having completed BN7 first.  Also grants The Blade's Simulacrum augment when starting BladeBurner if you have completed BN 7.

Sphyxis committed:
BB: Allow API Usage in BN 6 without 7 - Grants Augment for finishing BN 7

This allows the user to use the Netscript API in BN 6 without having completed BN7 first.  Also grants The Blade's Simulacrum augment when starting BladeBurner if you have completed BN 7.

Alpheus commented on src/NetscriptFunctions/Bladeburner.ts:
```diff
@@ -307,7 +307,7 @@ export function NetscriptBladeburner(): InternalAPI<INetscriptBladeburner> {
     },
     joinBladeburnerDivision: (ctx) => () => {
-      if (!canAccessBitNodeFeature(7) || Player.bitNodeOptions.disableBladeburner) {
+      if ((!canAccessBitNodeFeature(7) && !canAccessBitNodeFeature(6)) || Player.bitNodeOptions.disableBladeburner) {
```
Small suggestion:
I think it's worth in this case to split the if into two ifs:
- return false if BB is disabled
- return false if ~~!(no6 && no7)~~ (!no6 && !no7)

d0sboots replied:
+1, since these are just quick early-out false checks. (Also, the 2nd condition Alpheus suggested is wrong, so keep the current logic XD)

d0sboots commented:
Note: This is a continuation of #1407, PR redone due to merge conflicts (presumably).

Sphyxis committed:
Moved join check to 2 if statements

Broke out the checks for if you can access BN 6 or 7 and if the bitnode options are disabled for BladeBurner.

d0sboots commented on src/NetscriptFunctions/Bladeburner.ts:
```diff
@@ -307,7 +307,9 @@ export function NetscriptBladeburner(): InternalAPI<INetscriptBladeburner> {
-      if ((!canAccessBitNodeFeature(7) && !canAccessBitNodeFeature(6)) || Player.bitNodeOptions.disableBladeburner) {
+      if (!canAccessBitNodeFeature(7) && !canAccessBitNodeFeature(6)) {
+        return false; //Does not have bitnode 6 or 7
+      } else if (Player.bitNodeOptions.disableBladeburner) {
```
The else isn't needed, but that's OK.
